x-pack/plugin/sql/sql-action/src/main/java/org/elasticsearch/xpack/sql/action/AbstractSqlQueryRequest.java

```diff
@@ -117,6 +123,86 @@ public AbstractSqlQueryRequest params(List<SqlTypedParamValue> params) {
         this.params = params;
         return this;
     }
+    
+    private static List<SqlTypedParamValue> parseParams(XContentParser p) throws IOException {
+        List<SqlTypedParamValue> result = new ArrayList<>();
+        Token token = p.currentToken();
+        
+        if (token == Token.START_ARRAY) {
+            Object value = null;
+            String type = null;
+            SqlTypedParamValue previousParam = null;
+            
+            while ((token = p.nextToken()) != Token.END_ARRAY) {
+                XContentLocation loc = p.getTokenLocation();
+                
+                if (token == Token.START_OBJECT) {
+                    // we are at the start of a value/type pair... hopefully
+                    SqlTypedParamValue s = SqlTypedParamValue.fromXContent(p);
+                    /*
+                     * Set the xcontentlocation for the first param just in case the first one doesn't meet the parsing rules
```

Comment:
Was wondering if this was meant as `... for the first param that doesn't meet ...`, since an object could occur at any point in the array.

Reply:
I rephrased this section for, hopefully, better clarity.